### PR TITLE
DOC: Add F2PY tests documentation

### PR DESCRIPTION
**Pull request by NamamiShanker**
I have added a test suite documentation for F2PY

### CODE REVIEW

File: doc/source/f2py/index.rst [hunk 1]
```diff
@@ -36,6 +36,7 @@ replace all calls to ``f2py`` mentioned in this guide with the longer version.
 
    f2py-user
    f2py-reference
+   f2py-testing
```
Comment (HaoZeke) [inline doc/source/f2py/index.rst:39]: Looking at the rendered docs, perhaps the best place for this is under the Reference Guide --> `numpy/doc/source/f2py/f2py-reference.rst`

**Review by HaoZeke — COMMENTED** [review]
LGTM. Thanks for writing this up, reads very well, barring the minor comment this should be ready to go.

File: doc/source/f2py/index.rst (hunk 1 above)
Comment (NamamiShanker) [reply doc/source/f2py/index.rst]: Right, I changed it. Please have a look.

**Review by HaoZeke — APPROVED** [review]
Fantastic, LGTM. Thanks!

**Pull request merged**